assets/src/edit-story/components/checklist/checks/videoOptimization.js

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { __ } from '@web-stories-wp/i18n';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useCallback, useMemo, useState } from 'react';
+import { useLocalMedia, useStory } from '../../../app';
+import { VIDEO_SIZE_THRESHOLD } from '../../../app/media/utils/useFFmpeg';
+import { PRIORITY_COPY } from '../constants';
+import { LayerThumbnail, Thumbnail, THUMBNAIL_TYPES } from '../../thumbnail';
+import {
+  CARD_TYPE,
+  ChecklistCard,
+  DefaultFooterText,
+} from '../../checklistCard';
+import { filterStoryElements } from '../utils/filterStoryElements';
+import { Button, BUTTON_SIZES, BUTTON_TYPES } from '../../../../design-system';
+import { useHighlights } from '../../../app/highlights';
+
+const OptimizeButton = styled(Button)`
+  margin-top: 4px;
+  border: ${({ theme }) => `1px solid ${theme.colors.border.defaultNormal}`};
+`;
+
+export function videoElementOptimized(element = {}) {
+  if (element.resource?.isTranscoding) {
+    return true;
+  }
+
+  if (
+    element.type !== 'video' ||
+    element.resource?.local ||
+    element.resource?.isOptimized
+  ) {
+    return false;
+  }
+
+  const videoArea =
+    (element.resource?.height ?? 0) * (element.resource?.width ?? 0);
+  const isLargeVideo =
+    videoArea >= VIDEO_SIZE_THRESHOLD.WIDTH * VIDEO_SIZE_THRESHOLD.HEIGHT;
+  return isLargeVideo;
+}
+
+export const BulkVideoOptimization = () => {
+  const story = useStory(({ state }) => state);
+  const unoptimizedVideos = filterStoryElements(story, videoElementOptimized);
+
+  const setHighlights = useHighlights(({ setHighlights }) => setHighlights);
+
+  const { optimizeVideo } = useLocalMedia((state) => ({
+    optimizeVideo: state.actions.optimizeVideo,
+  }));
+
+  const [isOptimizing, setIsOptimizing] = useState(false);
+
+  const sequencedVideoOptimization = useCallback(() => {
+    return new Promise((res) => {
+      unoptimizedVideos
+        .filter(({ resource }) => !resource?.isTranscoding)
+        .map(({ resource }) => async () => {
+          await optimizeVideo({ resource });
+        })
+        .reduce((p, fn) => p.then(fn), Promise.resolve())
+        .then(() => {
+          res();
+        });
+    });
+  }, [unoptimizedVideos, optimizeVideo]);
+
+  const handleUpdateVideos = useCallback(async () => {
+    setIsOptimizing(true);
+    await sequencedVideoOptimization();
+    setIsOptimizing(false);
+  }, [sequencedVideoOptimization]);
+
+  const { footer, title } = PRIORITY_COPY.videoNotOptimized;
+
+  const isTranscoding = useMemo(
+    () =>
+      isOptimizing ||
+      unoptimizedVideos.some((video) => video.resource?.isTranscoding),
+    [isOptimizing, unoptimizedVideos]
+  );
+  return (
+    unoptimizedVideos.length > 0 && (
+      <ChecklistCard
+        cta={
```

Comment:
I think this CTA should only be present if there's more than 1 video, otherwise it's redundant. 
![Screen Shot 2021-06-25 at 9 29 59 AM](https://user-images.githubusercontent.com/10720454/123456755-0f196580-d598-11eb-803b-da38c5c6edb0.png)

This might actually navigate out of the issue where when you 'optimize all' on 1 video ya get an uncaught promise error. 
<img width="511" alt="Screen Shot 2021-06-25 at 9 33 25 AM" src="https://user-images.githubusercontent.com/10720454/123457071-661f3a80-d598-11eb-9e39-306f3b632ea9.png">